exoscale/datasource_exoscale_affinity.go

```diff
@@ -0,0 +1,73 @@
+package exoscale
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/exoscale/egoscale"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAffinity() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:          schema.TypeString,
+				Description:   "ID of the Affinity Group",
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+			"name": {
+				Type:          schema.TypeString,
+				Description:   "Name of the Affinity Group",
+				Optional:      true,
+				ConflictsWith: []string{"id"},
+			},
+		},
+
+		Read: dataSourceAffinityRead,
+	}
+}
+
+func dataSourceAffinityRead(d *schema.ResourceData, meta interface{}) error {
+	var err error
+
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutCreate))
+	defer cancel()
+
+	client := GetComputeClient(meta)
+
+	req := egoscale.ListAffinityGroups{Type: "host anti-affinity"}
```

Comment:
I think you can use [listAntiAffinityGroups](https://github.com/exoscale/egoscale/blob/master/antiaffinity_groups.go#L72) call ;)

Reply:
Good catch, I always forget that we have that one now 😉